src/lib.rs

```diff
@@ -537,7 +560,10 @@ impl Build {
             cmd.arg("-c");
         }
 
-        cmd.arg(&src);
+        cmd.arg(&src).stdout(Stdio::null());
+
+        // Add the flag to be tested
+        cmd.arg(flag);
 
         let output = cmd.output()?;
         let is_supported = output.status.success() && output.stderr.is_empty();
```

Comment:
If you're now converting warnings into errors, then the second half of this check is now unnecessary and ends up requiring additional handling like the `-v` stuff above and might yet still trigger false negatives.